sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1526,8 +1526,8 @@ class AdaptiveQueryExecSuite
         val dfRepartitionWithNum = df.repartition(5, 'b)
         dfRepartitionWithNum.collect()
         val planWithNum = dfRepartitionWithNum.queryExecution.executedPlan
-        // The top shuffle from repartition is optimized out.
-        assert(!hasRepartitionShuffle(planWithNum))
+        // The top shuffle from repartition is not optimized out.
```

Comment:
Ideally, when we optimize out the user-specified repartition (with num partitions), we should mark the operator that produces the same output partitioning as "not optimizable", so that we don't break the semantic of user-specified repartition.

However, this is super complicated and it doesn't seem to be worthwhile to optimize user-specified repartitions. I'm +1 to just not optimize out user-specified repartition in AQE.